src/test/scala/CoreServerTests.scala

```diff
@@ -0,0 +1,340 @@
+          case Success(verRes) => succeed
+          case Failure(_) => fail()
+        })
+      }
+
+      "be able to execute 'stop()' without exceptions" in {
+        core.stop()
+      }
+    }
+  }
+  "An instance of ViperCoreServer" when {
+    "verifying files with caching enabled" should {
+      val file1 = sumFile
+      val file2 = verifiableFile
+      val astGen1 = new AstGenerator(file1, silent_logger)
+      val astGen2 = new AstGenerator(file2, silent_logger)
+
+      val config = new ViperConfig(List())
+      config.verify()
+      val core = new ViperCoreServer(config)
+      core.start()
+
+      "be able to execute 'flushCache()' without exceptions after verifying files" in {
+        val handler1 = core.verify(file1, SiliconConfig(List()), astGen1.viper_ast.get)
+        core.flushCache()
+        val handler2 = core.verify(file2, SiliconConfig(List()), astGen2.viper_ast.get)
+        core.flushCache()
+        val handler3 = core.verify(file2, SiliconConfig(List()), astGen2.viper_ast.get)
+        core.flushCache()
+      }
+
+      "be able to execute 'stop()' without exceptions" in {
+        core.stop()
+      }
+    }
+  }
+  "An instance of ViperCoreServer" when {
+    "maximum capacity of verification jobs is exceeded" should {
+      val file = sumFile
+      val astGen = new AstGenerator(file, silent_logger)
+
+      val config = new ViperConfig(List())
+      config.verify()
+      val core = new ViperCoreServer(config)
+      core.start()
+
+      val handler = core.verify(file, SiliconConfig(List()), astGen.viper_ast.get)
+      core.verify(file, SiliconConfig(List()), astGen.viper_ast.get)
+      core.verify(file, SiliconConfig(List()), astGen.viper_ast.get)
+
+      "have 'verify()' return a VerificationJobHandler with negative id." in {
+        val spillHandler = core.verify(file, SiliconConfig(List()), astGen.viper_ast.get)
+        assert(spillHandler.id < 0)
+      }
+
+      "until a job handle has been freed." in {
+        val result_future = core.getMessagesFuture(handler.id)
+        while (!result_future.isCompleted) {
+          Thread.sleep(500)
+        }
+        val newHandler = core.verify(file, SiliconConfig(List()), astGen.viper_ast.get)
+        assert(newHandler.id > 0)
+      }
+
+      "be able to execute 'stop()' without exceptions" in {
+        core.stop()
+      }
+    }
+  }
+  "An instance of ViperCoreServer" when {
+    "verifying a file with caching disabled" should {
+      val file = sumFile
+      val astGen = new AstGenerator(file, silent_logger)
+
+      val config = new ViperConfig(List())
+      config.verify()
+      var core = new ViperCoreServer(config)
+      core.start()
+
+      var handler: VerificationJobHandler = null
+      handler = core.verify(file, SiliconConfig(List("--disableCaching")), astGen.viper_ast.get)
+
+      "be able to have 'streamMessages()' stream Viper messages to a client actor." in {
+        core.streamMessages(handler.id, client_actor)
```

Comment:
I wrote a suggestion about how to make this test case more meaningful. Currently it only tests that the actor system doesn't crash.